department-of-veterans-affairs/caseflow
# MattT/APPEALS-7465: Create/Update Seed Data for Caregiver Effort

Resolves [APPEALS-7465](https://vajira.max.gov/browse/APPEALS-7465)

# Description
Removes existing Caregiver PO from our test database seed. Adds the new VHA Caregiver Support organization with two users.

# Acceptance Criteria
- [ ] Create seed data for Caregiver organization and users
- [ ] Remove existing VHA Program Office

# Testing Plan
1. Seed new test data into database. You have two options:

Note: If you're testing this branch in a caseflowdemo.com instance then you can skip this step.

### The Fast Method:

Comment out lines 16, 18, and 19 (leave 17 uncommented) in [db/seeds/veterans_health_administration.rb](https://github.com/department-of-veterans-affairs/caseflow/blob/cb602b212ce4bb15a11a648ca7ba750721fe7ed7/db/seeds/veterans_health_administration.rb#L15)

Open the Rails console and paste the following commands one at a time:
```ruby
require_relative 'db/seeds/veterans_health_administration'

Seeds::VeteransHealthAdministration.new.seed!
```

Remember to uncomment the lines in `seed!`

### The Slow Method:

In the shell (not Rails console):
```bash
# In case you haven't already done this. If the Makefile already exists you can skip this.
ln -s Makefile.example Makefile

# Resets everything for safe measure
make reset
```

2. Navigate to `/test/users`

3. Make sure that `CAREGIVERUSER` and `CAREGIVERADMIN` are both in the dropdown. Switch to either one of those users.

4. Navigate to `/queue`. Switch to the CSP's queue:

<img width="852" alt="switch-to-csp-queue" src="https://user-images.githubusercontent.com/99351305/181835942-b5037c5f-de0b-4767-b8e6-9e928de78bd0.PNG">

5. Ensure that both users are a part of the CSP org and have the proper roles. In the Ruby console:
```ruby
# New Users
csp_user = User.find_by_css_id("CAREGIVERUSER")
csp_admin = User.find_by_css_id("CAREGIVERADMIN")

# New Org
# By not using VhaCaregiveSupport.singleton it'll tell us if the seed has correctly created the org or not
# .singleton creates the org if it doesn't already exist, which can obscure any possible failures in the seed file if used
vha_csp = VhaCaregiverSupport.first

# Check for org membership
# Expected results are notated as: => expected_result
vha_csp.users.include?(csp_user)
=> true

vha_csp.users.include?(csp_admin)
=> true

# Check if users are admins
vha_csp.user_is_admin?(csp_user)
=> false

vha_csp.user_is_admin?(csp_admin)
=> true
```

6. Make sure the old program office is gone:
```ruby
VhaProgramOffice.where(name: "Caregiver Appeals").count
=> 0
```
